ingestion/src/metadata/workflow/workflow_status_mixin.py

```diff
@@ -0,0 +1,80 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Add methods to the workflows for updating the IngestionPipeline status
+"""
+import uuid
+from datetime import datetime
+from typing import Optional
+
+from metadata.generated.schema.entity.services.ingestionPipelines.ingestionPipeline import (
+    PipelineState,
+    PipelineStatus,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    OpenMetadataWorkflowConfig,
+)
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+
+
+class WorkflowStatusMixin:
+    """
+    Helper methods to manage IngestionPipeline status
+    and workflow run ID
+    """
+
+    config: OpenMetadataWorkflowConfig
+    _run_id: Optional[str] = None
+    metadata: OpenMetadata
+
+    @property
+    def run_id(self) -> str:
```

Comment:
It's cleaner to handle the run id setting logic once instead of computing every status call.

Also, the old `set_ingestion_pipeline_status` was returning a `str` of the pipeline run id, which was misleading